models/migrations/v16.go

```diff
@@ -0,0 +1,117 @@
+package migrations
+
+import (
+	"fmt"
+	"time"
+
+	"code.gitea.io/gitea/modules/markdown"
+
+	"github.com/go-xorm/xorm"
+)
+
+// RepoUnit describes all units of a repository
+type RepoUnit struct {
+	ID          int64
+	RepoID      int64 `xorm:"INDEX(s)"`
+	Type        int   `xorm:"INDEX(s)"`
+	Index       int
+	Config      map[string]string `xorm:"JSON"`
```

Comment:
Would it be better to have the keys of `RepoUnit.Config` be an enum type instead of string literals? I see two advantages to using an enum:
1. We don't need to have "magic" string literals in our code. It is easy to accidentally mistype something like `"ExternalTrackerFormat"`, and the compiler won't tell you if you do.
2. When the `RepoUnit.Config` field is converted to JSON to store in the database, the generated JSON object will have shorter fields, and thus will use up less disk space.